packages/gatsby-telemetry/src/telemetry.ts

```diff
@@ -165,11 +165,19 @@ export class AnalyticsTracker {
   }
 
   // We might have two instances of this lib loaded, one from globally installed gatsby-cli and one from local gatsby.
-  // Hence we need to use process level globals that are not scoped to this module
+  // Hence we need to use process level globals that are not scoped to this module.
+  // Due to the forking on develop process, we also need to pass this via process.env so that child processes have the same sessionId
   getSessionId(): string {
     const p = process as any
     if (!p.gatsbyTelemetrySessionId) {
-      p.gatsbyTelemetrySessionId = uuidv4()
+      const inherited = process.env.INTERNAL_GATSBY_TELEMETRY_SESSION_ID
+      if (inherited) {
+        p.gatsbyTelemetrySessionId = inherited
+      } else {
+        p.gatsbyTelemetrySessionId = uuidv4()
+        process.env.INTERNAL_GATSBY_TELEMETRY_SESSION_ID =
+          p.gatsbyTelemetrySessionId
+      }
     }
```

Comment:
I didn't verify yet, but I think setting `process.env.*` only if `p.gatsbyTelemetrySessionId` is falsy will not handle the case when older global `gatsby-cli` is used (which will use older `gatsby-telemetry`) meaning that env var won't be set and child process will continue to generate new ids.

should there be extra condition like:

```js
if (!p.gatsbyTelemetrySessionId) {
  // [...] same as what's already in this PR
} else if (!process.env.INTERNAL_GATSBY_TELEMETRY_SESSION_ID) {
  // in case older `gatsby-telemetry` already set `gatsbyTelemetrySessionId` property on process
  // but didn't set env var - let's make sure env var is set
  process.env.INTERNAL_GATSBY_TELEMETRY_SESSION_ID = p.gatsbyTelemetrySessionId
}
```

Admiteddly, depending on order of things getting executed - this might also not be enough (in case we fork before there is some telemetry event that would use newer local `gatsby-telemetry`)

Reply:
@pieh is right. I initially tested with `npm run develop` which worked (there was a single `sessionId` in `events.json`)

Running gatsby develop (with a globally installed `gatsby-cli` doesn't work and sends multiple (3) sessionIds again

Reply:
nice catch! updated